lib.php

```diff
@@ -322,7 +322,7 @@ function reengagement_cron() {
                   INNER JOIN {reengagement} r on r.id = ip.reengagement
                        WHERE ip.emailtime < :emailtime
                              AND r.emailuser = " . REENGAGEMENT_EMAILUSER_TIME . '
-                             AND ip.emailsent = 0
+                             AND ip.emailsent <= r.remindercount
```

Comment:
I think you're out-by-one here:
if remindercount is 2,
and a user has been sent 2 emails, that record will still be caught.